sklearn/preprocessing/data.py

```diff
@@ -2028,7 +2028,11 @@ def fit(self, X, y=None):
 
         if y is not None:
             for i in range(len_data):
-                X_key = tuple(X.take(i, axis=0).take(inclusion_used).flatten())
+                X_key_1 = X.take(i, axis=0)
```

Comment:
tl;dr I want my code to be general enough to handle 1D and 2D arrays as inputs without any special handling, and I haven't really found any good solutions.